Add instructions on managing a separate mount namespace (#29653)
kubernetes/website: content/en/docs/tasks/administer-cluster/separate-mount-namespace.md

---
title: "Managing a Separate Mount Namespace for Kubernetes"
linkTitle: "Separate Mount Namespace"
weight: 100
description: >-

>      Discussion and instructions on how to administer a Kubernetes system with all container-specific mountpoints in a separate mount namespace from the parent system

Reviewer: ```suggestion
    Configure a Kubernetes node with all container-specific mountpoints
    in a separate mount namespace from the parent operating system.
```

---

> # Managing a Separate Mount Namespace for Kubernetes

Reviewer: The site renderer adds a heading automatically; we don't need this one here so I would omit it.
(to preview locally: `make container-image && make container-serve`, then visit http://localhost:1313/)

## What does this mean?

Originally, Kubelet and the Container Runtime all ran in the same mount namespace as
the host OS (systemd, login shells, system daemons).

![Fig 1: Original mount propagation](images/original-k8s-mount-propagation.png)

> _Fig 1: Original mount propagation_

Reviewer: nit: this would work well using a [figure](https://gohugo.io/content-management/shortcodes/#figure)

Author: I'll give that a shot - However, I didn't see my images in the local preview, just broken image links.  Do you know if I'm putting them in the wrong place?

However, in order to support the full range of cloud-native applications, only
two guarantees need to be made about the relationship of mountpoints:

1. Mounts originating in the host OS are visible to the Container Runtime, OCI
   hooks, Kubelet, and container volumeMounts with `MountPropagation:
   HostToContainer` (or `Bidirectional`)
2. Mounts originating in the Container Runtime, OCI hooks, Kubelet, and
   container volumeMounts with `mountPropagation: Bidirectional` are visible to
   each other and container volumeMounts with `MountPropagation:
   HostToContainer`

The fact that mounts originating in the Container Runtime, OCI hooks, Kubelet,
and container volumeMounts with `mountPropagation: Bidirectional` were visible
to the host OS in the original implementations of Kubernetes was incidental to
the implementation, and there are a number of real-world benefits to changing
the way Kubernetes is installed in order to hide the container-specific mount
points from the host OS.

> ![Fig 2: Container-Runtime-scoped mount propagation](images/hidden-k8s-mount-propagation.png)

Reviewer: nit: this would work well using a [figure](https://gohugo.io/content-management/shortcodes/#figure)
